### PR TITLE
SAML support

### PR DESCRIPTION
**Pull request by sdesen**

### CODE REVIEW

**Review by neelav — COMMENTED** [review]
I have no issues with this PR.  DB changes look good to me.  @aaronpk and @kamronbatman should be the final reviewers here.

File: authorization-server/routes/saml.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import makeConfiguration from '../server-configuration.js';
+import Account from '../utils/account.js';
+import { cacheSubjectToken } from '../utils/id-token-cache.js';
+import { createSAMLStrategy } from '../utils/passport-strategy.js';
+
+const body = urlencoded({ extended: false });
+
+export default async (app, provider) => {
+  // TODO: The OIDC resource mounts this route and fails
+  if (!process.env.CUSTOMER1_SAML_ISSUER || process.env.CUSTOMER1_SAML_ISSUER === '') {
```
Comment (kamronbatman) [inline authorization-server/routes/saml.js:13]: nit, empty string is falsy
```suggestion
  if (!process.env.CUSTOMER1_SAML_ISSUER) {
```

File: authorization-server/utils/id-token-cache.js [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
+/**
+ *
+ * Caches the ID Assertion after SSO for token exchange
+ *
+ * NOTE: In production, the cache lookup key should be the authorization_code
+ *
+ */
+
+function makeCacheKey(subject, client_id) {
+  return [subject, client_id].join('|');
```
Comment (kamronbatman) [inline authorization-server/utils/id-token-cache.js:12]: another nit, I am not a fan of array/join paradigm for just two elements. Up to you on this though:
```suggestion
  return `${subject}|${client_id}`;
```

File: authorization-server/utils/id-token-cache.js [hunk 3]
```diff
@@ -1,9 +1,23 @@
...
+
+export function cacheSubjectToken(subject, client_id, subjectToken, tokenType) {
+  cache[makeCacheKey(subject, client_id)] = { subjectToken, tokenType };
 }
 
-export function getIdToken(subject) {
-  return cache[subject];
+export function getSubjectToken(subject, client_id) {
+  return (
+    cache[makeCacheKey(subject, client_id)] || { subjectToken: undefined, tokenType: undefined }
```
Comment (kamronbatman) [inline authorization-server/utils/id-token-cache.js:21]: nit, should we use `??` here?

```suggestion
    cache[makeCacheKey(subject, client_id)] ?? { subjectToken: undefined, tokenType: undefined }
```

**Review by kamronbatman — APPROVED** [review]
This looks great! Just a few small nits that aren't material. Update as you see fit.

**Pull request merged**